### ISSUE TITLE
Invalid references are ignored

### ISSUE THREAD

**Issue by pylaligand** [rustless/valico#53]
```rust
use serde_json::json;
use serde_json::value::Value;
use valico::json_schema::Scope;

fn main() {
    let schema = json!({
        "properties": {
            "something": {
                "$ref": "nowhere_to_be_found.json#/definitions/some_type"
            }
        }
    });
    let mut scope = Scope::new();
    // Expecting a panic below since the reference cannot be resolved.
    let validator = scope.compile_and_return(schema, true).unwrap();

    // Anything is valid beyond this point.
    let validate = |file: &Value| {
        let result = validator.validate(file);
        println!("Valid? {}", result.is_valid());
    };

    validate(&json!({
        "something": "a random value"
    }));
    // Prints out "Valid? true".

    validate(&json!({
        "something": [ 1, 2 ,3 ]
    }));
    // Prints out "Valid? true".
}
```

I'd expect some error somewhere if a reference cannot be resolved. Is an empty schema being substituted instead?

**Comment by cramertj** [comment]
cc @s-panferov

**Comment by gklijs** [comment]
It's in the docs it's not working, https://github.com/s-panferov/valico/blob/master/README.md#json-schema but I wonder why, and if it could be fixed. Probably you need to build some kind of resolve context, and add the schema's with the name for all the references. If a reference is missing I think it should already give an error when compiling the schema? Or maybe it should stay optional, and have something like a 'validate_resolve_refs'?

**Comment by sbihel** [comment]
There simply is no HTTP client to fetch remote schemas. I might work on it soon.

There is already the concept of missing reference (which you can check with [`is_strictly_calid()`](https://docs.rs/valico/3.4.0/valico/json_schema/validators/struct.ValidationState.html#method.is_strictly_valid)).

**Comment by gklijs** [comment]
I didn't know about `is_strictly_valid()` that good enough for my use case. I do have a minor problem with references not necessary being fully qualified urls. When you use `scope.compile_with_id` the references are properly handled.
I guess it would be nice to have something like `reference foo could no be resolved` in the errors through. Currently when it's not strictly valid because of a missing reference, there is also no error, which is confusing.

**Comment by sbihel** [comment]
Yeah, I don't know if when compiling a schema it should check that the reference returns something. In a way, I feel like it shouldn't, it should just check that it is a valid URI-reference/JSON-pointer as it is kind of validating the schema against the JSON-Schema meta-schema. But then how do detect circular references? How do you make a nice user experience? How do you deal with remote refs changing (or becoming unavailable) over time? How do you deal with recursive schemas?

I think that's exactly what https://github.com/json-schema-org/json-schema-spec/issues/514 is about but I haven't read in detail.

**Comment by gklijs** [comment]
I think since there are many options it would be best for this library if it was just a string, without any validation. There are to many possible scenarios.

For example when create Java code from defenition files, the references need to be relative file paths. In my case the references can be fetched from Schema Registry later on, and resolved using this library.
